pkg/storegateway/bucket.go

```diff
@@ -257,8 +257,13 @@ func NewBucketStore(
 		option(s)
 	}
 
+	lazyLoaded := indexheader.LazyLoadedHeadersSnapshotConfig{
```

Comment:
`lazyLoaded` doesn't quite capture what this is - perhaps some variation of the type name would be clearer?